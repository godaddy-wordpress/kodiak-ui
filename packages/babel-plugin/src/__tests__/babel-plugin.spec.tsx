import babelTester from '../../scripts/babel-tester'
import plugin from '@kodiak-ui/babel-plugin'

// runs all files in the __fixtures__ folder through a babel transformation and snapshots them
babelTester('@emotion/babel-plugin', __dirname, {
  plugins: [plugin],
})
