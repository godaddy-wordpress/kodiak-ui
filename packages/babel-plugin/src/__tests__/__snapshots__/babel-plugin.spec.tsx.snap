// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/babel-plugin hoist 1`] = `
"import * as React from 'react'
import { Box } from '@kodiak-ui/primitives'

function sxGenerator() {
  return {
    bg: 'green',
  }
}

function App() {
  const isTrue = true

  return (
    <div
      sx={{
        backgroundColor: 'blue',
        '&:hover': {
          color: 'lightgreen',
        },
      }}
    >
      This has a hotpink background.
      <div
        css={{ backgroundColor: isTrue ? 'red' : 'blue' }}
        sx={{
          color: isTrue ? 'red' : 'blue',
        }}
        onClick={() => null}
      >
        Another div that isn't hoisted
      </div>
      <Box
        variants={['var1', 'var2']}
        sx={{ background: 'blue' }}
        aria-label=\\"a label\\"
      >
        A box
      </Box>
      <Box sx={sxGenerator()}>Pure functions are also hoisted</Box>
    </div>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import * as React from 'react';
import { Box } from '@kodiak-ui/primitives';

function sxGenerator() {
  return {
    bg: 'green'
  };
}

var _ref = {
  backgroundColor: 'blue',
  '&:hover': {
    color: 'lightgreen'
  }
};
var _ref2 = ['var1', 'var2'];
var _ref3 = {
  background: 'blue'
};

var _ref4 = sxGenerator();

function App() {
  const isTrue = true;
  return <div sx={_ref}>
      This has a hotpink background.
      <div css={{
      backgroundColor: isTrue ? 'red' : 'blue'
    }} sx={{
      color: isTrue ? 'red' : 'blue'
    }} onClick={() => null}>
        Another div that isn't hoisted
      </div>
      <Box variants={_ref2} sx={_ref3} aria-label=\\"a label\\">
        A box
      </Box>
      <Box sx={_ref4}>Pure functions are also hoisted</Box>
    </div>;
}"
`;
